src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/readers/JSONTemplateReader.java

```diff
@@ -135,7 +154,11 @@ private void getBuilderFromJsonArray(
             JsonNode node,
             TemplateBuilder currentBuilder,
             TemplateBuilderMaker maker) {
-        TemplateBuilder iteratingBuilder = maker.name(nodeName).collection(true).build();
+        TemplateBuilder iteratingBuilder =
+                maker.name(nodeName)
+                        .collection(true)
+                        .topLevelFeature(isRootOrManaged(currentBuilder))
```

Comment:
Confusing, as the maker also has a "managed" property.... an explanation is needed somewhere

Reply:
added some clarifying comments. The topLevelFeature builder will always be a children of a managed or of a RootBuilder.
That's why the method is checking for parent being RootOrManaged